### ISSUE TITLE
☂️ Add Swift Package Manager compatibility

### ISSUE THREAD

**Issue by vashworth** [flutter/flutter#126005]
## Add Swift Package Manager compatibility

Current status: 
* SwiftPM is currently available as a experimental feature: https://docs.flutter.dev/packages-and-plugins/swift-package-manager
* Iterating on SwiftPM by adding the Flutter framework as a local override: https://github.com/flutter/flutter/issues/166486

See Github subtasks for progress and planned work

----

### Tracking of additional features and bugs:
- [x] https://github.com/flutter/flutter/issues/153448
- [ ] https://github.com/flutter/flutter/issues/148424
- [x] https://github.com/flutter/flutter/issues/163086
- [x] https://github.com/flutter/flutter/issues/162594
- [ ] https://github.com/flutter/flutter/issues/157391
- [ ] https://github.com/flutter/flutter/issues/161182
- [ ] https://github.com/flutter/flutter/issues/165420
- [ ] https://github.com/flutter/flutter/issues/144397
  * Common SwiftPM errors are not currently parsed for macOS due to this issue. Once resolved, we should parse for [common errors](https://github.com/flutter/flutter/blob/7d891907e380afa826d2252aeb997c74a1b9292b/packages/flutter_tools/lib/src/ios/mac.dart#L894-L935) same as iOS
- [ ] https://github.com/flutter/flutter/issues/162196
- [x] https://github.com/flutter/flutter/issues/162399
- [ ] https://github.com/flutter/flutter/issues/163284
- [ ] https://github.com/flutter/flutter/issues/160329
- [ ] https://github.com/flutter/flutter/issues/146203
- [ ] [packages] Consider re-adding SwiftPM .modulemaps (see https://github.com/flutter/flutter/issues/148572#issuecomment-2231296570)
- [ ] https://github.com/dart-lang/pub/issues/4262
  * Workaround in documentation (using !.gitkeep in .gitignore)

## Resources
Design doc:
* [flutter.dev/go/swift-package-manager-plugins-v2](https://flutter.dev/go/swift-package-manager-plugins-v2) (outdated, see https://github.com/flutter/flutter/pull/146256)
* [flutter.dev/go/swift-package-manager-plugins-v3](https://flutter.dev/go/swift-package-manager-plugins-v3)

Original Github Issue: https://github.com/flutter/flutter/issues/33850

**Comment by renanyoy** [comment]
would be nice to to do it quicly, as it feels like it's the end of coacoapods.. tons of dead repos there...

**Comment by stuartmorgan** [comment]
@renanyoy Flutter does not support Swift as a language for any of those platforms. If you want to propose new supported languages that would need to be a new issue.

**Comment by dawid-niedzwiecki** [comment]
I am all for it :D

**Comment by danysz** [comment]
Any news on this ? We would be happy to move from cocoapods.

**Comment by importRyan** [comment]
This is exciting work. Please consider in the migration path that SPM will ignore architecture exclusions when a scheme includes "debug" or "development" in its name. Flutter requires "Debug" as a prefix.

**Comment by trivikr** [comment]
Updates from @orta of the CocoaPods team:

> TLDR: We're still keeping it ticking, but we're being more up-front that CocoaPods is in **maintenance mode**.

> If CocoaPods' only audience were native Cocoa developers, CocoaPods' usage should be on the decline, however, that is not the case. The popularity of React Native and Flutter have ensured that [most metrics of usage/traffic](https://www.ruby-toolbox.com/projects/cocoapods) have been steadily rising over time.

> **Q: I use CocoaPods as a Hidden Abstraction in My Framework**
> A: E.g. I maintain Unity, React Native, Flutter etc. A lot of these projects will (and should) be migrating to Swift Package Manager with time.

Sources:

* https://blog.cocoapods.org/CocoaPods-Support-Plans/
* https://x.com/orta/status/1823302734423965845

**Comment by scopendo** [comment]
I understand that we can enable SPM globally and then disable on a per project basis, however, from an app developer's point of view it would be helpful to be able to _enable_ SPM on a per project basis.

**Comment by stuartmorgan-g** [comment]
I believe Flutter 3.35 already supports the new general pubspec flag configuration syntax, and the Flutter SwiftPM docs just haven't been updated:

```yaml
flutter:
  config:
    enable-swift-package-manager: true
```
